pytorch_lightning/trainer/connectors/logger_connector/fx_validator.py

```diff
@@ -0,0 +1,91 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Dict, Optional, Tuple
+
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+
+class FxValidator:
+
+    def check_logging(self, fx_name: str, on_step: bool, on_epoch: bool) -> None:
+        allowed = self.functions[fx_name]
+        if allowed is None:
+            raise MisconfigurationException(f"{fx_name} function doesn't support logging using `self.log()`")
+
+        m = "You can't `self.log({}={})` inside `{}`, must be one of {}"
+        if on_step not in allowed["on_step"]:
+            msg = m.format("on_step", on_step, fx_name, allowed["on_step"])
+            raise MisconfigurationException(msg)
+
+        if on_epoch not in allowed["on_epoch"]:
+            msg = m.format("on_epoch", on_epoch, fx_name, allowed["on_epoch"])
+            raise MisconfigurationException(msg)
+
+    @property
+    def functions(self) -> Dict[str, Optional[Dict[str, Tuple[bool]]]]:
+        return dict(
```

Comment:
since this is called once per `self.log` would it make sense to store the dict instead of creating a new one each time?

Reply:
Yes. Changing it to a class attribute.